### PR TITLE
Added makefile to create virtual environment

### PR DESCRIPTION
**Pull request by ayuhsya**
Using the make command will install virtualenv if not installed and setup the venv in the project folder. Will also display messages to further activate the environment and run the sample code.

### CODE REVIEW

**Comment by abinashmeher999** [comment]
@ayuhsya Could you enable edits by maintainers? It's in the lower right corner. Thanks. I will review it soon.

**Comment by ayuhsya** [comment]
It's already enabled.

**Comment by abinashmeher999** [comment]
@ayuhsya Couple of things.
`pip` doesn't work without sudo on my machine so, we need a way to handle that. And also make activate doesn't seem to work. It just says `/bin/sh: source: not found`.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+TARGET: requirements.txt wiki_cat.py
+	@echo "First time install only"
+	pip install virtualenv
+	virtualenv -p /usr/bin/python3 venv
+	@echo "Activate virtual environment with source venv/bin/activate command"
+	@echo "To install dependencies, use pip install -r requirements.txt"
+	@echo "run sample code wiki_cat.py using python3 wiki_cat.py"
+
+activate:
```
Comment (abinashmeher999) [inline Makefile]: better if it checks for existence of `venv` first.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+TARGET: requirements.txt wiki_cat.py
```
Comment (abinashmeher999) [inline Makefile]: This need not check for `requirements.txt` or `wiki_cat.py`.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	@echo "To install dependencies, use pip install -r requirements.txt"
+	@echo "run sample code wiki_cat.py using python3 wiki_cat.py"
+
+activate:
+	( \
+		source ./venv/bin/activate; \
+		pip install -r requirements.txt; \
+	)
+	
+uninstall:
```
Comment (abinashmeher999) [inline Makefile:12]: This should check for `venv` too.

**Review by abinashmeher999 — CHANGES_REQUESTED** [review]
I have more suggestions to add which i will make myself. Meanwhile if you could change these it would be great. Also you can squash both the commits, \`Date` doesn't really convey much.

**Comment by ayuhsya** [comment]
@abinashmeher999 Hey, I've made the final changes, do I need to send another PR?

**Comment by ayuhsya** [comment]
@abinashmeher999 Also, I was kinda noob in using the squash command and squashed my commits to your latest commit :neutral_face: .

**Comment by abinashmeher999** [comment]
@ayuhsya I don't have issues with merging this. But just so you know, what you did can be undone using `git reflog`. Do you want to try?

**Comment by ayuhsya** [comment]
@abinashmeher999 Umm, if there are no issues with merging then I request you to merge it. It says in my fork that it's one commit ahead of your repository, so it's OK I think.

**Comment by abinashmeher999** [comment]
@ayuhsya Are you sure you have enabled edits. Because I keep getting this
```bash
git push ayushya slave
Username for 'https://github.com': abinashmeher999
Password for 'https://abinashmeher999@github.com': 
error: src refspec slave does not match any.
error: failed to push some refs to 'https://github.com/ayuhsya/wikipedia-category.git'
```

**Comment by ayuhsya** [comment]
Yes, I'm very positive the "Allow edits from maintainers." on the lower right corner is checked. Maybe try with +slave to force push? :neutral_face:

**Review by abinashmeher999 — APPROVED** [review]

**Comment by abinashmeher999** [comment]
@ayuhsya Great work! :tada: Giving the output in red was thoughtful :slightly_smiling_face:

**Comment by ayuhsya** [comment]
Why would I want to delete the branch? Do I have to make a new branch for every issue?

**Comment by abinashmeher999** [comment]
There's no need for the branch anymore because it has been merged with master. You can checkout a new branch from the latest master and start working on it.

**Comment by ayuhsya** [comment]
So then I have to merge my fork with the latest master?

**Comment by abinashmeher999** [comment]
Better yet rebase it.
```bash
git checkout master
git fetch upstream
git rebase upstream/master
git push origin master
```
This would work assuming your local master hasn't changed.

**Comment by abinashmeher999** [comment]
@ayuhsya An unfortunate result of that squashing incident is that your name now doesn't appear in the list of authors of commits. I am manually adding your name for now.

**Pull request merged**